### DISCUSSION TITLE
'Surface runoff' decisions

### DISCUSSION THREAD

**Discussion by omayou** [CH-Earth/summa#512] [General]
Hi, 

SUMMA's first publication (the horrendogram) presents three approaches for parametrizing 'surface runoff': (i) Explicit overland flow, (ii) Water table (TOPMODEL), and (iii) Xinanjiang (VIC). Unfortunately, I haven't been able to find detailed information about them in the publication, the online documentation (readthedocs.io), or the source codes (e.g., computFlux.F90). 
I intend to test each and introduce, if possible, new parameterizations (e.g., diffusive wave approximation of the Saint Venant equation and other semi-empirical equations) for peatland catchments.
Thank you so much.

Regards,
Behzad

**Comment by wknoben** [comment]
Hi,
Surface runoff is calculated in file `soilLiqFlx.f90` in subroutine `surfaceFlx()`. The magnitude of the surface runoff flux is calculated as `scalarSurfaceRunoff = scalarRainPlusMelt - scalarSurfaceInfiltration`. If you trace these calculations through the file you'll find how the magnitude of the infiltration flux is calculated. This in part depends on your choice of upper boundary conditions for the hydrology of the soil domain and may be the part of the code where you can trial alternative ways to calculate surface runoff.

As for the horrendogram, I think it's best to see that (as its caption says) as a conceptual outline of a framework like SUMMA rather than an accurate description of the code that is in SUMMA at any given point in time.
